{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4ff1d1db-9574-4009-a48a-65ea63b879b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import pprint\n",
    "sns.set(style=\"white\", context='paper')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1b5e4c3-f38f-4f41-80e7-94e489fce6db",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d5376016-6126-48a8-9e18-048386948111",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['.ipynb_checkpoints',\n",
       " 'CC1A3 HOT1A3 analyze EC.ipynb',\n",
       " 'CC1A3 HOT1A3 compare exo list.ipynb',\n",
       " 'chunks',\n",
       " 'data',\n",
       " 'generate_fasta_chunks.py',\n",
       " 'HOT1A3_protease_ROS_LUCA.csv',\n",
       " 'HOT1A3_ROS_protease.csv',\n",
       " 'signalP output',\n",
       " 'signalp.ipynb']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "dpath = '.'\n",
    "os.listdir(dpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "42418b27-2d9a-4a45-96ff-d35c7e10b307",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>Name</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_synonym</th>\n",
       "      <th>locus_tag</th>\n",
       "      <th>old_locus_tag</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>...</th>\n",
       "      <th>ecpath</th>\n",
       "      <th>Category</th>\n",
       "      <th>Membrane transport</th>\n",
       "      <th>kegg_gene1</th>\n",
       "      <th>kegg_gene2</th>\n",
       "      <th>SignalP</th>\n",
       "      <th>is_signal</th>\n",
       "      <th>EC_number_from_kegg</th>\n",
       "      <th>ROS_degradation</th>\n",
       "      <th>Protease_Aminase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NZ_CP012202.1</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS00010</td>\n",
       "      <td>ACZ81_00010</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>1625</td>\n",
       "      <td>2725</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>K02338  dnaN; DNA polymerase III subunit beta ...</td>\n",
       "      <td>Genetic Info</td>\n",
       "      <td>NaN</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>(dnaN)</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>2.7.7.7</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NZ_CP012202.1</td>\n",
       "      <td>recF</td>\n",
       "      <td>recF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS00015</td>\n",
       "      <td>ACZ81_00015</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>2851</td>\n",
       "      <td>3939</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>K03629  recF; DNA replication and repair prote...</td>\n",
       "      <td>Genetic Info</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recF</td>\n",
       "      <td>(recF)</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NZ_CP012202.1</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS00020</td>\n",
       "      <td>ACZ81_00020</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>3948</td>\n",
       "      <td>6368</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>K02470  gyrB; DNA gyrase subunit B [EC:5.6.2.2]</td>\n",
       "      <td>Genetic Info</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>(gyrB)</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>5.6.2.2</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NZ_CP012202.1</td>\n",
       "      <td>ACZ81_RS00025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS00025</td>\n",
       "      <td>ACZ81_00025</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>6540</td>\n",
       "      <td>6983</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPO</td>\n",
       "      <td>Is Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NZ_CP012202.1</td>\n",
       "      <td>glyS</td>\n",
       "      <td>glyS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS00030</td>\n",
       "      <td>ACZ81_00030</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>7069</td>\n",
       "      <td>9147</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>K01879  glyS; glycyl-tRNA synthetase beta chai...</td>\n",
       "      <td>AA/Nucleotide</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glyS</td>\n",
       "      <td>(glyS)</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>6.1.1.14</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4018</th>\n",
       "      <td>NZ_CP012203.1</td>\n",
       "      <td>ACZ81_RS20690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS20690</td>\n",
       "      <td>ACZ81_20690</td>\n",
       "      <td>GeneMarkS-2+</td>\n",
       "      <td>141226</td>\n",
       "      <td>141534</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SP</td>\n",
       "      <td>Is Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4019</th>\n",
       "      <td>NZ_CP012203.1</td>\n",
       "      <td>ACZ81_RS21705</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS21705</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>142282</td>\n",
       "      <td>142794</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4020</th>\n",
       "      <td>NZ_CP012203.1</td>\n",
       "      <td>ACZ81_RS20705</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS20705</td>\n",
       "      <td>ACZ81_20705</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>143592</td>\n",
       "      <td>144026</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>K13638  zntR; MerR family transcriptional regu...</td>\n",
       "      <td>Genetic Info</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4021</th>\n",
       "      <td>NZ_CP012203.1</td>\n",
       "      <td>ACZ81_RS20720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS20720</td>\n",
       "      <td>ACZ81_20720</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>146726</td>\n",
       "      <td>147064</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4022</th>\n",
       "      <td>NZ_CP012203.1</td>\n",
       "      <td>ACZ81_RS20730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS20730</td>\n",
       "      <td>ACZ81_20730</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>147705</td>\n",
       "      <td>148115</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SP</td>\n",
       "      <td>Is Signal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4023 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             seq_id           Name  gene gene_synonym      locus_tag  \\\n",
       "0     NZ_CP012202.1           dnaN  dnaN          NaN  ACZ81_RS00010   \n",
       "1     NZ_CP012202.1           recF  recF          NaN  ACZ81_RS00015   \n",
       "2     NZ_CP012202.1           gyrB  gyrB          NaN  ACZ81_RS00020   \n",
       "3     NZ_CP012202.1  ACZ81_RS00025   NaN          NaN  ACZ81_RS00025   \n",
       "4     NZ_CP012202.1           glyS  glyS          NaN  ACZ81_RS00030   \n",
       "...             ...            ...   ...          ...            ...   \n",
       "4018  NZ_CP012203.1  ACZ81_RS20690   NaN          NaN  ACZ81_RS20690   \n",
       "4019  NZ_CP012203.1  ACZ81_RS21705   NaN          NaN  ACZ81_RS21705   \n",
       "4020  NZ_CP012203.1  ACZ81_RS20705   NaN          NaN  ACZ81_RS20705   \n",
       "4021  NZ_CP012203.1  ACZ81_RS20720   NaN          NaN  ACZ81_RS20720   \n",
       "4022  NZ_CP012203.1  ACZ81_RS20730   NaN          NaN  ACZ81_RS20730   \n",
       "\n",
       "     old_locus_tag            source   start     end strand  ...  \\\n",
       "0      ACZ81_00010  Protein Homology    1625    2725      +  ...   \n",
       "1      ACZ81_00015  Protein Homology    2851    3939      +  ...   \n",
       "2      ACZ81_00020  Protein Homology    3948    6368      +  ...   \n",
       "3      ACZ81_00025  Protein Homology    6540    6983      -  ...   \n",
       "4      ACZ81_00030  Protein Homology    7069    9147      -  ...   \n",
       "...            ...               ...     ...     ...    ...  ...   \n",
       "4018   ACZ81_20690      GeneMarkS-2+  141226  141534      +  ...   \n",
       "4019           NaN  Protein Homology  142282  142794      +  ...   \n",
       "4020   ACZ81_20705  Protein Homology  143592  144026      +  ...   \n",
       "4021   ACZ81_20720  Protein Homology  146726  147064      +  ...   \n",
       "4022   ACZ81_20730  Protein Homology  147705  148115      +  ...   \n",
       "\n",
       "                                                 ecpath         Category  \\\n",
       "0     K02338  dnaN; DNA polymerase III subunit beta ...     Genetic Info   \n",
       "1     K03629  recF; DNA replication and repair prote...     Genetic Info   \n",
       "2       K02470  gyrB; DNA gyrase subunit B [EC:5.6.2.2]     Genetic Info   \n",
       "3                                                   NaN  Uncharacterized   \n",
       "4     K01879  glyS; glycyl-tRNA synthetase beta chai...    AA/Nucleotide   \n",
       "...                                                 ...              ...   \n",
       "4018                                                NaN  Uncharacterized   \n",
       "4019                                                NaN  Uncharacterized   \n",
       "4020  K13638  zntR; MerR family transcriptional regu...     Genetic Info   \n",
       "4021                                                NaN            Other   \n",
       "4022                                                NaN  Uncharacterized   \n",
       "\n",
       "     Membrane transport kegg_gene1 kegg_gene2  SignalP   is_signal  \\\n",
       "0                   NaN       dnaN     (dnaN)    OTHER  Not Signal   \n",
       "1                   NaN       recF     (recF)    OTHER  Not Signal   \n",
       "2                   NaN       gyrB     (gyrB)    OTHER  Not Signal   \n",
       "3                   NaN        NaN        NaN     LIPO   Is Signal   \n",
       "4                   NaN       glyS     (glyS)    OTHER  Not Signal   \n",
       "...                 ...        ...        ...      ...         ...   \n",
       "4018                NaN        NaN        NaN       SP   Is Signal   \n",
       "4019                NaN        NaN        NaN    OTHER  Not Signal   \n",
       "4020                NaN        NaN        NaN    OTHER  Not Signal   \n",
       "4021                NaN        NaN        NaN    OTHER  Not Signal   \n",
       "4022                NaN        NaN        NaN       SP   Is Signal   \n",
       "\n",
       "     EC_number_from_kegg      ROS_degradation      Protease_Aminase  \n",
       "0                2.7.7.7  Not ROS degradation  Not Protease/Aminase  \n",
       "1                    NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "2                5.6.2.2  Not ROS degradation  Not Protease/Aminase  \n",
       "3                    NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "4               6.1.1.14  Not ROS degradation  Not Protease/Aminase  \n",
       "...                  ...                  ...                   ...  \n",
       "4018                 NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "4019                 NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "4020                 NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "4021                 NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "4022                 NaN  Not ROS degradation  Not Protease/Aminase  \n",
       "\n",
       "[4023 rows x 30 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df = pd.read_csv('10cc_ecpath_master.csv')\n",
    "ncbi_df = pd.read_csv('HOT1A3_ROS_protease.csv')\n",
    "ncbi_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e60df4c4-1282-4cc0-85f8-84ac7200499c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>locus_tag</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>product</th>\n",
       "      <th>db_xref</th>\n",
       "      <th>EC_number</th>\n",
       "      <th>...</th>\n",
       "      <th>category_x</th>\n",
       "      <th>N Enzyme name</th>\n",
       "      <th>EC numbers</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>_merge</th>\n",
       "      <th>EC_number_pattern</th>\n",
       "      <th>category_y</th>\n",
       "      <th>enzyme name</th>\n",
       "      <th>category</th>\n",
       "      <th>is_signal</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00264</td>\n",
       "      <td>CDS</td>\n",
       "      <td>288697</td>\n",
       "      <td>290727</td>\n",
       "      <td>-</td>\n",
       "      <td>ftsH_1</td>\n",
       "      <td>ATP-dependent zinc metalloprotease FtsH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00278</td>\n",
       "      <td>CDS</td>\n",
       "      <td>306026</td>\n",
       "      <td>306970</td>\n",
       "      <td>-</td>\n",
       "      <td>htpX</td>\n",
       "      <td>Protease HtpX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00426</td>\n",
       "      <td>CDS</td>\n",
       "      <td>482986</td>\n",
       "      <td>484956</td>\n",
       "      <td>+</td>\n",
       "      <td>macB</td>\n",
       "      <td>Macrolide export ATP-binding/permease protein ...</td>\n",
       "      <td>COG:COG0577</td>\n",
       "      <td>3.6.3.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Membrane transporters</td>\n",
       "      <td>Membrane transporters</td>\n",
       "      <td>3.6.3.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Membrane transporters</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00472</td>\n",
       "      <td>CDS</td>\n",
       "      <td>530621</td>\n",
       "      <td>532729</td>\n",
       "      <td>-</td>\n",
       "      <td>dpp5_1</td>\n",
       "      <td>Dipeptidyl-peptidase 5</td>\n",
       "      <td>COG:COG1506</td>\n",
       "      <td>3.4.14.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Du/tripeptidyl-peptidases</td>\n",
       "      <td>3.4.14.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00600</td>\n",
       "      <td>CDS</td>\n",
       "      <td>663168</td>\n",
       "      <td>664613</td>\n",
       "      <td>+</td>\n",
       "      <td>ftsH_2</td>\n",
       "      <td>ATP-dependent zinc metalloprotease FtsH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4361</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_04232</td>\n",
       "      <td>CDS</td>\n",
       "      <td>146363</td>\n",
       "      <td>146620</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4362</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_04233</td>\n",
       "      <td>CDS</td>\n",
       "      <td>146726</td>\n",
       "      <td>147064</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4363</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_04234</td>\n",
       "      <td>CDS</td>\n",
       "      <td>147083</td>\n",
       "      <td>147721</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4364</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_04235</td>\n",
       "      <td>CDS</td>\n",
       "      <td>147705</td>\n",
       "      <td>148115</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4365</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_04236</td>\n",
       "      <td>CDS</td>\n",
       "      <td>148133</td>\n",
       "      <td>148246</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4366 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      filename       locus_tag type   start     end strand    gene  \\\n",
       "0     28108.53  IHCOGGJP_00264  CDS  288697  290727      -  ftsH_1   \n",
       "1     28108.53  IHCOGGJP_00278  CDS  306026  306970      -    htpX   \n",
       "2     28108.53  IHCOGGJP_00426  CDS  482986  484956      +    macB   \n",
       "3     28108.53  IHCOGGJP_00472  CDS  530621  532729      -  dpp5_1   \n",
       "4     28108.53  IHCOGGJP_00600  CDS  663168  664613      +  ftsH_2   \n",
       "...        ...             ...  ...     ...     ...    ...     ...   \n",
       "4361  28108.53  IHCOGGJP_04232  CDS  146363  146620      +     NaN   \n",
       "4362  28108.53  IHCOGGJP_04233  CDS  146726  147064      +     NaN   \n",
       "4363  28108.53  IHCOGGJP_04234  CDS  147083  147721      +     NaN   \n",
       "4364  28108.53  IHCOGGJP_04235  CDS  147705  148115      +     NaN   \n",
       "4365  28108.53  IHCOGGJP_04236  CDS  148133  148246      +     NaN   \n",
       "\n",
       "                                                product      db_xref  \\\n",
       "0               ATP-dependent zinc metalloprotease FtsH          NaN   \n",
       "1                                         Protease HtpX          NaN   \n",
       "2     Macrolide export ATP-binding/permease protein ...  COG:COG0577   \n",
       "3                                Dipeptidyl-peptidase 5  COG:COG1506   \n",
       "4               ATP-dependent zinc metalloprotease FtsH          NaN   \n",
       "...                                                 ...          ...   \n",
       "4361                               hypothetical protein          NaN   \n",
       "4362                               hypothetical protein          NaN   \n",
       "4363                               hypothetical protein          NaN   \n",
       "4364                               hypothetical protein          NaN   \n",
       "4365                               hypothetical protein          NaN   \n",
       "\n",
       "     EC_number  ...              category_x               N Enzyme name  \\\n",
       "0     3.4.24.-  ...               Proteases      Metalloendopeptidases    \n",
       "1     3.4.24.-  ...               Proteases      Metalloendopeptidases    \n",
       "2      3.6.3.-  ...  Membrane transporters       Membrane transporters    \n",
       "3     3.4.14.-  ...               Proteases  Du/tripeptidyl-peptidases    \n",
       "4     3.4.24.-  ...               Proteases      Metalloendopeptidases    \n",
       "...        ...  ...                     ...                         ...   \n",
       "4361       NaN  ...                     NaN                         NaN   \n",
       "4362       NaN  ...                     NaN                         NaN   \n",
       "4363       NaN  ...                     NaN                         NaN   \n",
       "4364       NaN  ...                     NaN                         NaN   \n",
       "4365       NaN  ...                     NaN                         NaN   \n",
       "\n",
       "      EC numbers Unnamed: 3     _merge EC_number_pattern  category_y  \\\n",
       "0       3.4.24.x        NaN       both               NaN         NaN   \n",
       "1       3.4.24.x        NaN       both               NaN         NaN   \n",
       "2        3.6.3.x        NaN       both               NaN         NaN   \n",
       "3       3.4.14.x        NaN       both               NaN         NaN   \n",
       "4       3.4.24.x        NaN       both               NaN         NaN   \n",
       "...          ...        ...        ...               ...         ...   \n",
       "4361         NaN        NaN  left_only               NaN         NaN   \n",
       "4362         NaN        NaN  left_only               NaN         NaN   \n",
       "4363         NaN        NaN  left_only               NaN         NaN   \n",
       "4364         NaN        NaN  left_only               NaN         NaN   \n",
       "4365         NaN        NaN  left_only               NaN         NaN   \n",
       "\n",
       "      enzyme name                category   is_signal  \n",
       "0             NaN               Proteases  Not Signal  \n",
       "1             NaN               Proteases  Not Signal  \n",
       "2             NaN  Membrane transporters   Not Signal  \n",
       "3             NaN               Proteases   Is Signal  \n",
       "4             NaN               Proteases  Not Signal  \n",
       "...           ...                     ...         ...  \n",
       "4361          NaN                     NaN  Not Signal  \n",
       "4362          NaN                     NaN  Not Signal  \n",
       "4363          NaN                     NaN  Not Signal  \n",
       "4364          NaN                     NaN   Is Signal  \n",
       "4365          NaN                     NaN  Not Signal  \n",
       "\n",
       "[4366 rows x 43 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df = pd.read_csv('10cc_ecpath_master.csv')\n",
    "luca_df = pd.read_csv('HOT1A3_protease_ROS_LUCA.csv')\n",
    "luca_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2ba1cebc-2955-48cf-a72b-9ee31f0b12f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ncbi_df = ncbi_df.loc[ncbi_df.Protease_Aminase.isin(['Protease/Aminase'])].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8616d319-50c5-4812-a85a-94a23bf90bc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Proteases', 'Membrane transporters ',\n",
       "       'Uptake, reduction and assimilation', 'electron sources or sinks',\n",
       "       'Other enzymes relevant to N cycling', 'Transaminases',\n",
       "       'D-/L-Amino acid oxidases', nan, 'ROS'], dtype=object)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luca_df.category.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "43b90c69-0a49-4756-ba75-22b3521eb911",
   "metadata": {},
   "outputs": [],
   "source": [
    "#luca_df = luca_df.loc[luca_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases'])].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "64402aa1-ab21-42cf-a4b0-5f732cc1c418",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>locus_tag</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>product</th>\n",
       "      <th>db_xref</th>\n",
       "      <th>EC_number</th>\n",
       "      <th>...</th>\n",
       "      <th>category_x</th>\n",
       "      <th>N Enzyme name</th>\n",
       "      <th>EC numbers</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>_merge</th>\n",
       "      <th>EC_number_pattern</th>\n",
       "      <th>category_y</th>\n",
       "      <th>enzyme name</th>\n",
       "      <th>category</th>\n",
       "      <th>is_signal</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00264</td>\n",
       "      <td>CDS</td>\n",
       "      <td>288697</td>\n",
       "      <td>290727</td>\n",
       "      <td>-</td>\n",
       "      <td>ftsH_1</td>\n",
       "      <td>ATP-dependent zinc metalloprotease FtsH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00278</td>\n",
       "      <td>CDS</td>\n",
       "      <td>306026</td>\n",
       "      <td>306970</td>\n",
       "      <td>-</td>\n",
       "      <td>htpX</td>\n",
       "      <td>Protease HtpX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00472</td>\n",
       "      <td>CDS</td>\n",
       "      <td>530621</td>\n",
       "      <td>532729</td>\n",
       "      <td>-</td>\n",
       "      <td>dpp5_1</td>\n",
       "      <td>Dipeptidyl-peptidase 5</td>\n",
       "      <td>COG:COG1506</td>\n",
       "      <td>3.4.14.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Du/tripeptidyl-peptidases</td>\n",
       "      <td>3.4.14.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00600</td>\n",
       "      <td>CDS</td>\n",
       "      <td>663168</td>\n",
       "      <td>664613</td>\n",
       "      <td>+</td>\n",
       "      <td>ftsH_2</td>\n",
       "      <td>ATP-dependent zinc metalloprotease FtsH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00805</td>\n",
       "      <td>CDS</td>\n",
       "      <td>906181</td>\n",
       "      <td>908307</td>\n",
       "      <td>+</td>\n",
       "      <td>dpp5_2</td>\n",
       "      <td>Dipeptidyl-peptidase 5</td>\n",
       "      <td>COG:COG1506</td>\n",
       "      <td>3.4.14.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Du/tripeptidyl-peptidases</td>\n",
       "      <td>3.4.14.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00983</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1105303</td>\n",
       "      <td>1106655</td>\n",
       "      <td>+</td>\n",
       "      <td>rseP</td>\n",
       "      <td>Regulator of sigma-E protease RseP</td>\n",
       "      <td>COG:COG0750</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_01086</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1220537</td>\n",
       "      <td>1222612</td>\n",
       "      <td>-</td>\n",
       "      <td>pepO_1</td>\n",
       "      <td>Neutral endopeptidase</td>\n",
       "      <td>COG:COG3590</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_01876</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2129439</td>\n",
       "      <td>2131304</td>\n",
       "      <td>-</td>\n",
       "      <td>sppA</td>\n",
       "      <td>Protease 4</td>\n",
       "      <td>COG:COG0616</td>\n",
       "      <td>3.4.21.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Serine endopeptidases</td>\n",
       "      <td>3.4.21.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_01901</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2156579</td>\n",
       "      <td>2158525</td>\n",
       "      <td>+</td>\n",
       "      <td>ftsH_3</td>\n",
       "      <td>ATP-dependent zinc metalloprotease FtsH</td>\n",
       "      <td>COG:COG0465</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_02006</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2287276</td>\n",
       "      <td>2288322</td>\n",
       "      <td>+</td>\n",
       "      <td>sohB</td>\n",
       "      <td>putative protease SohB</td>\n",
       "      <td>COG:COG0616</td>\n",
       "      <td>3.4.21.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Serine endopeptidases</td>\n",
       "      <td>3.4.21.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_02244</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2543008</td>\n",
       "      <td>2545098</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Asp/Glu-specific dipeptidyl-peptidase</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.14.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Du/tripeptidyl-peptidases</td>\n",
       "      <td>3.4.14.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_02545</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2890437</td>\n",
       "      <td>2892614</td>\n",
       "      <td>-</td>\n",
       "      <td>dapb1</td>\n",
       "      <td>Dipeptidyl aminopeptidase BI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.4.14.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Du/tripeptidyl-peptidases</td>\n",
       "      <td>3.4.14.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_02664</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3018799</td>\n",
       "      <td>3020484</td>\n",
       "      <td>+</td>\n",
       "      <td>nadB</td>\n",
       "      <td>L-aspartate oxidase</td>\n",
       "      <td>COG:COG0029</td>\n",
       "      <td>1.4.3.16</td>\n",
       "      <td>...</td>\n",
       "      <td>D-/L-Amino acid oxidases</td>\n",
       "      <td>L-Aspartate oxidase</td>\n",
       "      <td>1.4.3.16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D-/L-Amino acid oxidases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_03214</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3656416</td>\n",
       "      <td>3657672</td>\n",
       "      <td>-</td>\n",
       "      <td>pepE_2</td>\n",
       "      <td>putative dipeptidase PepE</td>\n",
       "      <td>COG:COG0006</td>\n",
       "      <td>3.4.13.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Dipeptidases</td>\n",
       "      <td>3.4.13.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_03662</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4213216</td>\n",
       "      <td>4215282</td>\n",
       "      <td>+</td>\n",
       "      <td>pepO_2</td>\n",
       "      <td>Neutral endopeptidase</td>\n",
       "      <td>COG:COG3590</td>\n",
       "      <td>3.4.24.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Metalloendopeptidases</td>\n",
       "      <td>3.4.24.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Is Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_04120</td>\n",
       "      <td>CDS</td>\n",
       "      <td>62581</td>\n",
       "      <td>63135</td>\n",
       "      <td>-</td>\n",
       "      <td>hybD</td>\n",
       "      <td>Hydrogenase 2 maturation protease</td>\n",
       "      <td>COG:COG0680</td>\n",
       "      <td>3.4.23.-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Aspartic endopeptidases</td>\n",
       "      <td>3.4.23.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>28108.53</td>\n",
       "      <td>IHCOGGJP_00065</td>\n",
       "      <td>CDS</td>\n",
       "      <td>70565</td>\n",
       "      <td>71428</td>\n",
       "      <td>+</td>\n",
       "      <td>glpG</td>\n",
       "      <td>Rhomboid protease GlpG</td>\n",
       "      <td>COG:COG0705</td>\n",
       "      <td>3.4.21.105</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Serine endopeptidases</td>\n",
       "      <td>3.4.21.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>3.4.21.-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    filename       locus_tag type    start      end strand    gene  \\\n",
       "0   28108.53  IHCOGGJP_00264  CDS   288697   290727      -  ftsH_1   \n",
       "1   28108.53  IHCOGGJP_00278  CDS   306026   306970      -    htpX   \n",
       "3   28108.53  IHCOGGJP_00472  CDS   530621   532729      -  dpp5_1   \n",
       "4   28108.53  IHCOGGJP_00600  CDS   663168   664613      +  ftsH_2   \n",
       "9   28108.53  IHCOGGJP_00805  CDS   906181   908307      +  dpp5_2   \n",
       "12  28108.53  IHCOGGJP_00983  CDS  1105303  1106655      +    rseP   \n",
       "13  28108.53  IHCOGGJP_01086  CDS  1220537  1222612      -  pepO_1   \n",
       "19  28108.53  IHCOGGJP_01876  CDS  2129439  2131304      -    sppA   \n",
       "20  28108.53  IHCOGGJP_01901  CDS  2156579  2158525      +  ftsH_3   \n",
       "22  28108.53  IHCOGGJP_02006  CDS  2287276  2288322      +    sohB   \n",
       "25  28108.53  IHCOGGJP_02244  CDS  2543008  2545098      +     NaN   \n",
       "27  28108.53  IHCOGGJP_02545  CDS  2890437  2892614      -   dapb1   \n",
       "28  28108.53  IHCOGGJP_02664  CDS  3018799  3020484      +    nadB   \n",
       "32  28108.53  IHCOGGJP_03214  CDS  3656416  3657672      -  pepE_2   \n",
       "34  28108.53  IHCOGGJP_03662  CDS  4213216  4215282      +  pepO_2   \n",
       "37  28108.53  IHCOGGJP_04120  CDS    62581    63135      -    hybD   \n",
       "38  28108.53  IHCOGGJP_00065  CDS    70565    71428      +    glpG   \n",
       "\n",
       "                                    product      db_xref   EC_number  ...  \\\n",
       "0   ATP-dependent zinc metalloprotease FtsH          NaN    3.4.24.-  ...   \n",
       "1                             Protease HtpX          NaN    3.4.24.-  ...   \n",
       "3                    Dipeptidyl-peptidase 5  COG:COG1506    3.4.14.-  ...   \n",
       "4   ATP-dependent zinc metalloprotease FtsH          NaN    3.4.24.-  ...   \n",
       "9                    Dipeptidyl-peptidase 5  COG:COG1506    3.4.14.-  ...   \n",
       "12       Regulator of sigma-E protease RseP  COG:COG0750    3.4.24.-  ...   \n",
       "13                    Neutral endopeptidase  COG:COG3590    3.4.24.-  ...   \n",
       "19                               Protease 4  COG:COG0616    3.4.21.-  ...   \n",
       "20  ATP-dependent zinc metalloprotease FtsH  COG:COG0465    3.4.24.-  ...   \n",
       "22                   putative protease SohB  COG:COG0616    3.4.21.-  ...   \n",
       "25    Asp/Glu-specific dipeptidyl-peptidase          NaN    3.4.14.-  ...   \n",
       "27             Dipeptidyl aminopeptidase BI          NaN    3.4.14.-  ...   \n",
       "28                      L-aspartate oxidase  COG:COG0029    1.4.3.16  ...   \n",
       "32                putative dipeptidase PepE  COG:COG0006    3.4.13.-  ...   \n",
       "34                    Neutral endopeptidase  COG:COG3590    3.4.24.-  ...   \n",
       "37        Hydrogenase 2 maturation protease  COG:COG0680    3.4.23.-  ...   \n",
       "38                   Rhomboid protease GlpG  COG:COG0705  3.4.21.105  ...   \n",
       "\n",
       "                  category_x               N Enzyme name  EC numbers  \\\n",
       "0                  Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "1                  Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "3                  Proteases  Du/tripeptidyl-peptidases     3.4.14.x   \n",
       "4                  Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "9                  Proteases  Du/tripeptidyl-peptidases     3.4.14.x   \n",
       "12                 Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "13                 Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "19                 Proteases      Serine endopeptidases     3.4.21.x   \n",
       "20                 Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "22                 Proteases      Serine endopeptidases     3.4.21.x   \n",
       "25                 Proteases  Du/tripeptidyl-peptidases     3.4.14.x   \n",
       "27                 Proteases  Du/tripeptidyl-peptidases     3.4.14.x   \n",
       "28  D-/L-Amino acid oxidases        L-Aspartate oxidase     1.4.3.16   \n",
       "32                 Proteases               Dipeptidases     3.4.13.x   \n",
       "34                 Proteases      Metalloendopeptidases     3.4.24.x   \n",
       "37                 Proteases    Aspartic endopeptidases     3.4.23.x   \n",
       "38                 Proteases      Serine endopeptidases     3.4.21.x   \n",
       "\n",
       "   Unnamed: 3 _merge EC_number_pattern  category_y  enzyme name  \\\n",
       "0         NaN   both               NaN         NaN          NaN   \n",
       "1         NaN   both               NaN         NaN          NaN   \n",
       "3         NaN   both               NaN         NaN          NaN   \n",
       "4         NaN   both               NaN         NaN          NaN   \n",
       "9         NaN   both               NaN         NaN          NaN   \n",
       "12        NaN   both               NaN         NaN          NaN   \n",
       "13        NaN   both               NaN         NaN          NaN   \n",
       "19        NaN   both               NaN         NaN          NaN   \n",
       "20        NaN   both               NaN         NaN          NaN   \n",
       "22        NaN   both               NaN         NaN          NaN   \n",
       "25        NaN   both               NaN         NaN          NaN   \n",
       "27        NaN   both               NaN         NaN          NaN   \n",
       "28        NaN   both               NaN         NaN          NaN   \n",
       "32        NaN   both               NaN         NaN          NaN   \n",
       "34        NaN   both               NaN         NaN          NaN   \n",
       "37        NaN   both               NaN         NaN          NaN   \n",
       "38        NaN   both          3.4.21.-         NaN          NaN   \n",
       "\n",
       "                    category   is_signal  \n",
       "0                  Proteases  Not Signal  \n",
       "1                  Proteases  Not Signal  \n",
       "3                  Proteases   Is Signal  \n",
       "4                  Proteases  Not Signal  \n",
       "9                  Proteases   Is Signal  \n",
       "12                 Proteases  Not Signal  \n",
       "13                 Proteases   Is Signal  \n",
       "19                 Proteases  Not Signal  \n",
       "20                 Proteases  Not Signal  \n",
       "22                 Proteases  Not Signal  \n",
       "25                 Proteases  Not Signal  \n",
       "27                 Proteases   Is Signal  \n",
       "28  D-/L-Amino acid oxidases  Not Signal  \n",
       "32                 Proteases  Not Signal  \n",
       "34                 Proteases   Is Signal  \n",
       "37                 Proteases  Not Signal  \n",
       "38                 Proteases  Not Signal  \n",
       "\n",
       "[17 rows x 43 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luca_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "43e9d21f-8154-4a4a-9dbb-5fa92146904c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59      rhomboid family intramembrane serine protease ...\n",
       "226                                  metalloprotease TldD\n",
       "229                                  metalloprotease PmbA\n",
       "333                            M3 family metallopeptidase\n",
       "342                         penicillin-binding protein 1B\n",
       "350                             gamma-glutamyltransferase\n",
       "383               PBP1A family penicillin-binding protein\n",
       "461                             gamma-glutamyltransferase\n",
       "508                                   Xaa-Pro dipeptidase\n",
       "734      trypsin-like peptidase domain-containing protein\n",
       "735                       DegQ family serine endoprotease\n",
       "945                       sigma E protease regulator RseP\n",
       "1047                          M13 family metallopeptidase\n",
       "1132                            insulinase family protein\n",
       "1786                        signal peptide peptidase SppA\n",
       "1812              ATP-dependent zinc metalloprotease FtsH\n",
       "1918                                        protease SohB\n",
       "2086                           M48 family metalloprotease\n",
       "2131                                        rhombosortase\n",
       "2413                            insulinase family protein\n",
       "2476                            insulinase family protein\n",
       "2547                                  L-aspartate oxidase\n",
       "2584                                     endopeptidase La\n",
       "2586    ATP-dependent Clp endopeptidase proteolytic su...\n",
       "2684                                           exosortase\n",
       "2766                                     class GN sortase\n",
       "2843           TIGR02281 family clan AA aspartic protease\n",
       "3275                           S8 family serine peptidase\n",
       "3281                           S8 family serine peptidase\n",
       "3366                           S8 family serine peptidase\n",
       "3426                  HslU--HslV peptidase ATPase subunit\n",
       "3508                          M13 family metallopeptidase\n",
       "3935    HyaD/HybD family hydrogenase maturation endope...\n",
       "Name: product, dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_df['product']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7557f95f-26b6-487e-9cb0-bdb860458872",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_df = pd.merge(ncbi_df, luca_df, on=['start', 'end', 'strand'], how='outer', suffixes=['_ncbi', '_luca'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "2131464f-e3fd-4bb6-943c-8eb1a73e687d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Not ROS degradation', nan, 'ROS_degradation'], dtype=object)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_df.ROS_degradation.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "503ab099-6a4f-46ee-88a3-8bd633cc8039",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cols = [ 'locus_tag_ncbi','locus_tag_luca', 'gene_ncbi', 'gene_luca',\n",
    "       'product_ncbi', 'product_luca', #'is_signal_ncbi',  'is_signal_luca', \n",
    "        'Protease_Aminase',  \n",
    "           'category','EC_number', 'EC_number_from_kegg', 'is_signal_ncbi', 'is_signal_luca',\n",
    "       ]\n",
    "\n",
    "with pd.ExcelWriter('protease_comparison_ncbi_luca.xlsx') as writer:  \n",
    "    merge_df.loc[\n",
    "        merge_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases']) & \n",
    "        merge_df.Protease_Aminase.isin(['Protease/Aminase']), \n",
    "    cols].to_excel(writer, sheet_name='Common', index=False)\n",
    "\n",
    "    merge_df.loc[\n",
    "        ~merge_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases']) & \n",
    "        merge_df.Protease_Aminase.isin(['Protease/Aminase']), \n",
    "    cols].to_excel(writer, sheet_name='NCBI only', index=False)\n",
    "\n",
    "    merge_df.loc[\n",
    "        merge_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases']) & \n",
    "        ~merge_df.Protease_Aminase.isin(['Protease/Aminase']), \n",
    "    cols].to_excel(writer, sheet_name='LUCA only', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "e3ff8217-52d8-4654-b442-dbbdc28ae5af",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cols = [ 'locus_tag_ncbi','locus_tag_luca', 'gene_ncbi', 'gene_luca',\n",
    "       'product_ncbi', 'product_luca', #'is_signal_ncbi',  'is_signal_luca', \n",
    "        'Protease_Aminase',  \n",
    "           'category','EC_number', 'EC_number_from_kegg', 'is_signal_ncbi', 'is_signal_luca',\n",
    "       ]\n",
    "\n",
    "with pd.ExcelWriter('protease_comparison_ncbi_luca.xlsx') as writer:  \n",
    "    merge_df.loc[\n",
    "        merge_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases']) & \n",
    "        merge_df.Protease_Aminase.isin(['Protease/Aminase']), \n",
    "    cols].to_excel(writer, sheet_name='Common', index=False)\n",
    "\n",
    "    merge_df.loc[\n",
    "        ~merge_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases']) & \n",
    "        merge_df.Protease_Aminase.isin(['Protease/Aminase']), \n",
    "    cols].to_excel(writer, sheet_name='NCBI only', index=False)\n",
    "\n",
    "    merge_df.loc[\n",
    "        merge_df.category.isin(['Proteases', 'D-/L-Amino acid oxidases']) & \n",
    "        ~merge_df.Protease_Aminase.isin(['Protease/Aminase']), \n",
    "    cols].to_excel(writer, sheet_name='LUCA only', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "d9642bb2-7576-4bf3-94c8-33a08a51b785",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cols = [ 'locus_tag_ncbi','locus_tag_luca', 'gene_ncbi', 'gene_luca',\n",
    "       'product_ncbi', 'product_luca', #'is_signal_ncbi',  'is_signal_luca', \n",
    "        'Protease_Aminase',  \n",
    "           'category','EC_number', 'EC_number_from_kegg', 'is_signal_ncbi', 'is_signal_luca',\n",
    "       ]\n",
    "\n",
    "with pd.ExcelWriter('ROS_comparison_ncbi_luca.xlsx') as writer:  \n",
    "    merge_df.loc[\n",
    "        merge_df.category.isin(['ROS']) & \n",
    "        merge_df.ROS_degradation.isin(['ROS_degradation']), \n",
    "    cols].to_excel(writer, sheet_name='Common', index=False)\n",
    "\n",
    "    merge_df.loc[\n",
    "        ~merge_df.category.isin(['ROS']) & \n",
    "        merge_df.ROS_degradation.isin(['ROS_degradation']), \n",
    "    cols].to_excel(writer, sheet_name='NCBI only', index=False)\n",
    "\n",
    "    merge_df.loc[\n",
    "        merge_df.category.isin(['ROS']) & \n",
    "        ~merge_df.ROS_degradation.isin(['ROS_degradation']), \n",
    "    cols].to_excel(writer, sheet_name='LUCA only', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "7d5f4f97-1b5e-4169-bc10-86d8367edf41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>Name</th>\n",
       "      <th>gene_ncbi</th>\n",
       "      <th>gene_synonym</th>\n",
       "      <th>locus_tag_ncbi</th>\n",
       "      <th>old_locus_tag</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>...</th>\n",
       "      <th>category_x</th>\n",
       "      <th>N Enzyme name</th>\n",
       "      <th>EC numbers</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>_merge</th>\n",
       "      <th>EC_number_pattern</th>\n",
       "      <th>category_y</th>\n",
       "      <th>enzyme name</th>\n",
       "      <th>category</th>\n",
       "      <th>is_signal_luca</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3839</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3656416</td>\n",
       "      <td>3657672</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Dipeptidases</td>\n",
       "      <td>3.4.13.x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>both</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Proteases</td>\n",
       "      <td>Not Signal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     seq_id Name gene_ncbi gene_synonym locus_tag_ncbi old_locus_tag source  \\\n",
       "3839    NaN  NaN       NaN          NaN            NaN           NaN    NaN   \n",
       "\n",
       "        start      end strand  ... category_x  N Enzyme name EC numbers  \\\n",
       "3839  3656416  3657672      -  ...  Proteases  Dipeptidases    3.4.13.x   \n",
       "\n",
       "     Unnamed: 3 _merge  EC_number_pattern category_y enzyme name   category  \\\n",
       "3839        NaN   both                NaN        NaN         NaN  Proteases   \n",
       "\n",
       "     is_signal_luca  \n",
       "3839     Not Signal  \n",
       "\n",
       "[1 rows x 70 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_df.loc[merge_df.locus_tag_luca =='IHCOGGJP_03214']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "aa3693b7-ee03-4536-9a00-34ff133e3cf4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>Name</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_synonym</th>\n",
       "      <th>locus_tag</th>\n",
       "      <th>old_locus_tag</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>...</th>\n",
       "      <th>ecpath</th>\n",
       "      <th>Category</th>\n",
       "      <th>Membrane transport</th>\n",
       "      <th>kegg_gene1</th>\n",
       "      <th>kegg_gene2</th>\n",
       "      <th>SignalP</th>\n",
       "      <th>is_signal</th>\n",
       "      <th>EC_number_from_kegg</th>\n",
       "      <th>ROS_degradation</th>\n",
       "      <th>Protease_Aminase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3079</th>\n",
       "      <td>NZ_CP012202.1</td>\n",
       "      <td>ACZ81_RS15870</td>\n",
       "      <td>(pepQ)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACZ81_RS15870</td>\n",
       "      <td>ACZ81_15865</td>\n",
       "      <td>Protein Homology</td>\n",
       "      <td>3656416</td>\n",
       "      <td>3657621</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>K01271  pepQ; Xaa-Pro dipeptidase [EC:3.4.13.9]</td>\n",
       "      <td>Metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pepQ</td>\n",
       "      <td>(pepQ)</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not Signal</td>\n",
       "      <td>3.4.13.9</td>\n",
       "      <td>Not ROS degradation</td>\n",
       "      <td>Not Protease/Aminase</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             seq_id           Name    gene gene_synonym      locus_tag  \\\n",
       "3079  NZ_CP012202.1  ACZ81_RS15870  (pepQ)          NaN  ACZ81_RS15870   \n",
       "\n",
       "     old_locus_tag            source    start      end strand  ...  \\\n",
       "3079   ACZ81_15865  Protein Homology  3656416  3657621      -  ...   \n",
       "\n",
       "                                               ecpath    Category  \\\n",
       "3079  K01271  pepQ; Xaa-Pro dipeptidase [EC:3.4.13.9]  Metabolism   \n",
       "\n",
       "     Membrane transport kegg_gene1 kegg_gene2  SignalP   is_signal  \\\n",
       "3079                NaN       pepQ     (pepQ)    OTHER  Not Signal   \n",
       "\n",
       "     EC_number_from_kegg      ROS_degradation      Protease_Aminase  \n",
       "3079            3.4.13.9  Not ROS degradation  Not Protease/Aminase  \n",
       "\n",
       "[1 rows x 30 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_df.loc[ncbi_df.start.between(3656416-100, 3656416+100)& ncbi_df.end.between(3657672-100, 3657672+100)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "125f2e43-6b41-48cc-9061-d3674183140d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
